### DISCUSSION TITLE
export getDefaultConnectors

### DISCUSSION THREAD

**Discussion by q20274982** [family/connectkit#286] [Ideas]
When I attempting to provide a custom Connector that extends wagmi's `Connector` to the `createConfig.connectors` within connectkit, [the custom Connector replaces all the defaultConnectors](https://github.com/family/connectkit/blob/e2a9d2d77447143ec58822ced55b1fa66f05be8e/packages/connectkit/src/defaultConfig.ts#L192). Is it feasible to export the getDefaultConnectors function? This way, we could achieve a more flexible integration between the defaultConnectors and the custom Connector

**Comment by lochie** [comment]
Hey @q20274982, you can currently extend (or filter) the exported connectors like this:
```ts
const ckConfig = getDefaultConfig({
  ... // your config
})

ckConfig.connectors = [
  ...ckConfig.connectors,
  // your additional connector 
];

const config = createConfig(ckConfig); // wagmi's createConfig
```

**Reply by q20274982** [reply]
Hey @lochie , thank you for your response. Exporting `getDefaultConnectors` also takes a specific aspect into consideration.
When the `chains` parameter of `defaultConfig` is processed through `configureChains`, it might have deviated from the originally passed `chains` parameter. This discrepancy could lead to variations between the `configuredChains` within `defaultConfig` and the chains provided to a custom connector. Consequently, an additional step of utilizing `configureChains` to generate the chains configuration becomes necessary. This complexity of configuration further escalates when incorporating provider setups such as `alchemyProvider` or `infuraProvider`

```javascript
// before export getDefaultConnectors

const ckConfig = getDefaultConfig({
  ... // your config
})

// developers might inadvertently forget to configure `alchemyProvider` while passing alchemyId in `getDefaultConfig` when setting up configureChains's `providers`
const { chains: configuredChains  } = configureChains(chains, providers, { stallTimeout });

ckConfig.connectors = [
  ...ckConfig.connectors,
new myCustomConnector({ configuredChains })
];

const config = createConfig(ckConfig); // wagmi's createConfig

// after export getDefaultConnectors
// when users require a custom Connector, they have the option to configure wagmi themselves while simultaneously utilizing the `defaultConnectors` set up with connectkit

const {
  publicClient: configuredPublicClient,
  chains: configuredChains,
  webSocketPublicClient: configuredWebSocketPublicClient,
} = configureChains(chains,[alchemyProvider({ key }), publicProvider()], { stallTimeout });

const config = createConfig({
  // other options
  connectors: [
    new myCustomConnector({ configuredChains }),
    ...getDefaultConnectors({ configuredChains, appConfig, projectId })
  ],
  configuredPublicClient,
});
```

**Reply by lochie** [reply]
Excellent point, I'll have a think about this one a bit more before committing to anything just yet, but you're welcome to submit a PR.

It's also worth mentioning that you don't _have_ to use the `useDefaultConfig` function to generate your config and you can opt for writing your own following the [wagmi docs](https://wagmi.sh/react/config). The function we provide is just a helper that speeds up the process (and adds some nice-to-haves within the UX).

**Reply by q20274982** [reply]
Sure. I will initiate another PR.

Regarding connectors, I understand that developers can achieve the same effect as connectkit's `getDefaultConnectors` by manually configuring the default connectors offered by wagmi. However, the advantage of exporting `getDefaultConnectors` is that it reduces the developer's need to tediously configure commonly used default connectors themselves. At the same time, after adding a custom connector, it aims to maintain an experience as close as possible to the default connectkit setup.
